CoreArrangementPlugins/src/au/gov/asd/tac/constellation/plugins/arrangements/GraphTaxonomy.java

```diff
@@ -98,7 +98,7 @@ public GraphTaxonomy(final GraphWriteMethods wg, final Map<Integer, Set<Integer>
     }
 
     public GraphTaxonomy(final GraphWriteMethods wg, final Map<Integer, Set<Integer>> taxa, final Map<Integer, Integer> nodeToTaxa) {
-        this(wg, taxa, null, Graph.NOT_FOUND, Graph.NOT_FOUND);
+        this(wg, taxa, nodeToTaxa, Graph.NOT_FOUND, Graph.NOT_FOUND);
```

Comment:
just curious to know what this change addressed as I would assume this would alter functionality

Reply:
This constructor is only referenced in two classes TaxFromTrees and TaxFromNeighbours, both classes go through a lot of effort to generate the nodeToTaxa map and then this constructor simply ignores it. It appears to only effect the Tree Arrangement Plugins which are not overly performant anyway. their functionality is effected by this change however it doesn't become any better or worse, just different. So figured it would be better to assign the variable than to begin unpacking the methods that invoke the constructor (these methods only ever add to the map meaning that removing the parameter would result in code smells in the invoking methods where writing to a map that is never read from would need to be addressed). Happy to remove the value from the constructor and chase up subsequent code smells if you prefer

Reply:
Ok I've taken a closer look at `GraphTaxonomy`. Looking at it, it actually makes no sense for that particular constructor to be parsing null and instead and it should be parsing `nodeToTaxa` -> else there is no point in having it as a parameter (so that does seem like a bug). Curious to know exactly how the behaviour changes but worse case if the behaviour needs to be reverted, it would actually be better simply then using the 2 parameter constructor in those 2 cases as that makes `nodeToTaxa` null anyway (so achieves the exact same thing)